### DISCUSSION TITLE
Entity Framework 5 on SQLite

### DISCUSSION THREAD

**Discussion by bricelam** [bricelam/bricelam.github.io#79] [General]
# Entity Framework 5 on SQLite

This walkthrough will get you started with an application that uses the Entity Framework (EF) to read and write data from a SQLite database. It is intended to be similar to the Code First to a New Database walkthrough.

https://www.bricelam.net/2012/10/entity-framework-on-sqlite.html

**Comment by bricelam** [comment]
_From **Anonymous** on December 7, 2012 17:54_

<p>Very good sample :-)<br><br>Initially had a problem getting this sample to work. A quick Google lead me to the below stackoverflow.<br><br><br><a href="http://stackoverflow.com/questions/4225908/error-when-adding-a-configuration-to-app-config-file" rel="nofollow noopener" title="http://stackoverflow.com/questions/4225908/error-when-adding-a-configuration-to-app-config-file">http://stackoverflow.com/qu...</a></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Adwen** on December 22, 2012 2:14_

<p>Does this sqlite version work with entity framework 6 alpha ? I cannot get it to work.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on December 28, 2012 23:05_

<p>@Adwen, Not yet, but the System.Data.SQLite provider is open source and updating the provider to work with EF6 is as trivial as following instructions for <a href="http://entityframework.codeplex.com/wikipage?title=Rebuilding%20EF%20providers%20for%20EF6" rel="nofollow noopener" title="http://entityframework.codeplex.com/wikipage?title=Rebuilding%20EF%20providers%20for%20EF6">Rebuilding EF providers for EF6</a>.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Jason and Kris Carter** on January 10, 2013 6:24_

<p>Hey @Brice Lambson, I followed that tutorial, but I get the following exception when trying to use EF6.<br><br>The 'Instance' member of the Entity Framework provider type 'System.Data.SQLite.SQLiteFactory, System.Data.SQLite, Version=1.0.83.0, Culture=neutral, PublicKeyToken=db937bc2d44ff139' did not return an object that inherits from 'System.Data.Entity.Core.Common.DbProviderServices'. Entity Framework providers must extend from this class and the 'Instance' member must return the Singleton instance of the provider.<br><br>Is there some quick wrapper I can write for this, or is it a completely different provider architecture that SQLite will have to provide?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on January 13, 2013 6:32_

<p>I generated a POCO classes from SQL Server using DB-first (contained in their own project library).  My SQLite DB has the exact same schema, so I was hoping to reuse those POCO classes.  I would think this could work, but EF keeps trying to create the SQLite database and it is throwing a "UnintentionalCodeFirstException" error from OnModelCreating.  <br><br>I cannot figure out how to stop EF from trying to create the database and have tried "faking" metadata in the connection string as well as setting the  context element attribute disableDatabaseInitialization="true".<br><br>The SQLite database is set to copy to the output directory (if newer) and I have confirmed it exists under bin\Debug.<br><br>Not a big deal, but it is kind of frustrating that I seem unable to keep EF's magic from spreading its pixie dust when I don't want to fly!</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on January 14, 2013 22:08_

<p>@Anonymous, it sounds like you're mixing the database-first and Code First workflows. You should be able to reuse the POCO classes, but you will need to create a new DbContext from scratch -- not try and reuse the one generated using database-first.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on January 14, 2013 22:11_

<p>@Jason/Kris Carter, This sounds a bit like you're trying to use an EF5 provider with EF6. In order to use providers with EF6, they need <a href="http://entityframework.codeplex.com/wikipage?title=Rebuilding%20EF%20providers%20for%20EF6" rel="nofollow noopener" title="http://entityframework.codeplex.com/wikipage?title=Rebuilding%20EF%20providers%20for%20EF6">to be updated first</a> by the provider writers.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on January 16, 2013 17:17_

<p>I would like to use a database in a specific directory, i.e. D:\Data\MyDb.sqlite.<br>If I put that pathname in the connectionStrings in App.config, I'm able to read data from the database table, but new entities are not saved into its tables after context.SaveChanges().<br>How can I override this problem?<br>Thanks.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Никита Мулюкин** on January 28, 2013 1:15_

<p>My sample:<br>var post = new Post {Content = "test"};<br>            var post2 = new Post { Content = "test2"};<br><br>            try<br>            {<br>                using (var context = new TestContext())<br>                {<br>                    var sql = "create table Posts(Id int identity(1,1) primary key not null,Content varchar(1200))";<br>                    context.Database.ExecuteSqlCommand(sql);<br>                }<br>            }<br>            catch (Exception ex)<br>            {<br>                Console.WriteLine(ex.ToString());<br>            }<br><br>            try<br>            {<br>                using (var context = new TestContext())<br>                {<br>                    context.Posts.Add(post);<br>                    context.Posts.Add(post2);<br><br>                    context.SaveChanges();<br>                }<br>            }<br>            catch (Exception ex)<br>            {<br>                Console.WriteLine(ex.ToString());<br>            }<br><br>and i receive an error:<br>System.Data.Entity.Infrastructure.DbUpdateException: An error occurred while updating the entries. See the inner exception for details. ---&gt; System.Data.UpdateException: An error occurred while updating the entries. See the inner exception for details. ---&gt; System.Data.SQLite.SQLiteException: SQL logic error or missing database<br>no such table: Posts<br>   at System.Data.SQLite.SQLite3.Prepare(SQLiteConnection cnn, String strSql, SQLiteStatement previous, UInt32 timeoutMS, String&amp; strRemain)<br>   at System.Data.SQLite.SQLiteCommand.BuildNextCommand()<br><br>Post is:<br><br>public class Post<br>    {<br>        [Key]<br>        public int Id { get; set; }<br>        public string Content { get; set; }<br>    }<br><br>what i doing wrong? sorry from my english :(</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on January 28, 2013 22:21_

<p>@Никита, You need to create the database <em>before</em> using the context.<br><br>Create the table outside of the context like this:<br><br>using (var connection = new SQLiteConnection(myConnectionString))<br>using (var command = connection.CreateCommand())<br>{<br>    command.CommandText = "create table Posts(Id int identity(1,1) primary key not null,Content varchar(1200))";<br><br>    connection.Open();<br>    command.ExecuteNonQuery();<br>}</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on February 25, 2013 21:58_

<p>Thanks for this post!<br><br>This was my favorite - as I need to use EF wherever possible on free dbs and on free providers - like SQLite or Postgres with dotConnect Express.<br><br>Can you please direct me how can I use this further to set the data in objects as a datasource for Grid control? I want to be able to modify grid data and save it with context.SaveChanges(). I used so far the data adapter but it's really messy code!<br><br>Hopefully its not too much to ask!<br><br><br><br><br><br><br></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on February 25, 2013 22:56_

<p>@Anonymous, check out <a href="http://msdn.microsoft.com/en-us/jj682076" rel="nofollow noopener" title="http://msdn.microsoft.com/en-us/jj682076">Databinding with WinForms</a>.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on February 25, 2013 23:22_

<p>Very nice Brice,<br><br>I dont know how can I thank enough.<br><br>This page is my must go to page now onwards!<br><br></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Paul Ruiz Pauker** on February 26, 2013 2:43_

<p>Hi @Brice, nice post, it helped me a lot, but now I'm stuck on Model First aproach with SQLite, could you give me some lights on how to generate database model?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on February 26, 2013 5:06_

<p>@Paul, Unfortunately, System.Data.SQLite doesn't yet support DDL generation (see <a href="http://system.data.sqlite.org/index.html/tktview?name=c8083867b9" rel="nofollow noopener" title="http://system.data.sqlite.org/index.html/tktview?name=c8083867b9">Ticket c8083867b9</a>). I would consider using <a href="http://www.devart.com/dotconnect/sqlite/" rel="nofollow noopener" title="http://www.devart.com/dotconnect/sqlite/">Devart's provider</a> instead.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Paul Ruiz Pauker** on February 28, 2013 21:04_

<p>Thanks for the reply!. <br><br>Finally I've found another way of solving this, and, in case anyone else is looking for the same, I mean: "EF To Sqlite, model first", it's in:<br><br><a href="http://code.msdn.microsoft.com/Demo-of-ADONET-POCO-with-140ad3ad" rel="nofollow noopener" title="http://code.msdn.microsoft.com/Demo-of-ADONET-POCO-with-140ad3ad">http://code.msdn.microsoft....</a><br><br></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **myCollections myCollections** on April 10, 2013 18:55_

<p>This comment has been removed by the author.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **myCollections myCollections** on April 10, 2013 18:58_

<p>Hello Brice,<br><br>I just update sqlite provider as you described here :<br><br><a href="http://stackoverflow.com/questions/14510096/entity-framework-6-sqlite/15926681#15926681" rel="nofollow noopener" title="http://stackoverflow.com/questions/14510096/entity-framework-6-sqlite/15926681#15926681">http://stackoverflow.com/qu...</a><br><br>But i get the same error message as Jason :<br><br>"Entity Framework providers must extend from this class and the 'Instance' member must return the Singleton instance of the provider."<br><br>Could you help us :)<br><br>Thanks a lot for your time.<br><br>Regards<br></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on April 10, 2013 20:00_

<p>@myCollections myCollections,<br><br>Ensure you've removed all references to System.Data.Entity.dll from the provider and your project.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **myCollections myCollections** on April 11, 2013 10:11_

<p>Hello,<br><br>Thanks for your answer.<br>I have already removed System.Data.Entity.dll from System.Data.SQLite.Linq, System.Data.SQLite and my project.<br><br>Added EF6 nightly Build, rebuild both project using .net 4.5.<br><br>But without any success :(</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **HR** on April 18, 2013 11:49_

<p>This comment has been removed by the author.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **HR** on April 18, 2013 11:50_

<p>What if my code is in a dll (not exe) and I can't modify the exe's app.config?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **أحمد** on May 8, 2013 0:39_

<p>Really great article :) </p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on June 5, 2013 22:15_

<p>Hi,<br>I copied exacly what you wrote in the article. <br><br>SQL logic error or missing database no such table: Artist<br><br>What am I doing wrong?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on June 5, 2013 22:36_

<p>@Anonymous, make sure that the database file is being copied to the bin folder, and that the connection string filename matches the one on disk.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on June 6, 2013 0:14_

<p>I missed it.<br><br>Now it works, thanks!<br><br></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on June 12, 2013 17:54_

<p>Thanks a lot for this post man!!!</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Ryan Stowasser** on June 18, 2013 19:40_

<p>Thank you for this article, this has worked great for me until now. <br><br>I'm creating some reports from a sqlite database that doesn't use the naming scheme of AlbumId for and Album object. How do I specify a different field to load an object? An example would be Tickets.AssignedTo is an id that needs to load a User object?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on June 18, 2013 20:48_

<p>@Ryan Stowasser, I'm not entirely sure how you want your classes to look, but I'll do my best to reply.<br><br>You'll need to configure the relationship by overriding OnModelCreating in your DbContext and adding code similar to the following.<br><br>modelBuilder.Entity&lt;Ticket&gt;().HasOptional(t =&gt; t.AssignedToUser).WithMany(u =&gt; u.Tickets).HasForeignKey(t =&gt; t.AssignedTo);<br><br>For additional information, see <a href="http://msdn.microsoft.com/en-us/data/jj591620" rel="nofollow noopener" title="http://msdn.microsoft.com/en-us/data/jj591620">Configuring Relationships with the Fluent API</a>.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Ryan Stowasser** on June 19, 2013 3:03_

<p>Thank you Brice, that was very helpful to me.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on June 19, 2013 14:55_

<p>Hi Brice, this is a very interesting topic!<br><br>I'm not that familiar with Sqlite and entity framework, so I was wondering how it is done within a WPF application (data binding..)<br><br>Thanks in advance!</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on June 19, 2013 19:21_

<p>@Anonymous, here is an article on Entity Framework and <a href="http://msdn.microsoft.com/en-us/data/jj574514" rel="nofollow noopener" title="http://msdn.microsoft.com/en-us/data/jj574514">Databinding with WPF</a>.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Sylvia** on June 21, 2013 4:04_

<p>Hi,I have a question,I do the same thing with this article,but it show error "could not load sqlite.interop.dll",I installed the system.data.sqlite(x86 and x64) using PMD in vistual studio 2012,could you help me</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on June 21, 2013 23:36_

<p>@Sylvia, I hit the same issue when writing this post; that's why I suggest using the System.Data.SQLite<i>.x86</i> package.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Alex** on July 9, 2013 8:34_

<p>Hi i keep getting "context has changed since the database was created" error. Whats the equivalent of text field for ef</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Alex** on July 9, 2013 17:07_

<p>This comment has been removed by the author.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Alex** on July 9, 2013 17:08_

<p>Oh forget that last comment i was still connecting to my SqlDB and I had turned of the database initializer. I still cant figure why the connection isn't working though. I made all the connection strings and i can even access it via the Server Explorer.<br><br>is there anyway to post my app.config to see what the problem is?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Bill** on July 10, 2013 22:26_

<p>I have a working set up for NHibernate 3 and SQLite and everything is working exactly the way I want:  create an empty database, create the tables, populate the database, etc.<br><br>Now I want to do the same thing with EF5.  Using your tutorial as a starting point, I can get a new database created, but I cannot figure out how to get EF5 to create the tables for me.  This should be just as transparent as NHIbernate, as I am using the same SQLite driver.  <br><br>What do you suggest?<br><br>Thanks,<br>Bill</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **butteredtoast** on July 10, 2013 23:09_

<p>I am having troubles referencing the sample database using a Web API interface (verses the console application). I am using the select method in my GET function and I am getting a "SQL logic error or missing database no such table: Artist" error. Any ideas?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on July 11, 2013 0:09_

<p>@Bill, as I mentoned to Paul above, "Unfortunately, System.Data.SQLite doesn't yet support DDL generation (see <a href="http://system.data.sqlite.org/index.html/tktview?name=c8083867b9" rel="nofollow noopener" title="http://system.data.sqlite.org/index.html/tktview?name=c8083867b9">Ticket c8083867b9</a>). I would consider using <a href="http://www.devart.com/dotconnect/sqlite/" rel="nofollow noopener" title="http://www.devart.com/dotconnect/sqlite/">Devart's provider</a> instead."<br><br>@butteredtoast, Make sure your connection string is what you expect it to be at runtime (use db.Database.Connection.ConnectionString). If it uses "|DataDirectory|" make sure the database file is inside the App_Data directory.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on August 23, 2013 11:25_

<p>I'm currently using SQL Express and EF "database first", but am considering switching to SQLite. <br><br>Does SQLite support database first? And do you know of any tools that would let me export my SQL database (schema and data) into a SQLite database?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Brice Lambson** on August 23, 2013 19:17_

<p>@Anonymous As far as I know, it supports database first. You'll need to <a href="http://system.data.sqlite.org/index.html/doc/trunk/www/downloads.wiki" rel="nofollow noopener" title="http://system.data.sqlite.org/index.html/doc/trunk/www/downloads.wiki">download the SQLite design-time components for Visual Studio</a>. I'm not sure about any migration tools, but you might try asking on <a href="http://StackOverflow.com" rel="nofollow noopener" title="StackOverflow.com">StackOverflow.com</a> or <a href="http://ServerFault.com" rel="nofollow noopener" title="ServerFault.com">ServerFault.com</a>.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Unknown** on September 10, 2013 20:16_

<p>I spent a LOT of time getting this working in a class library as a plug-in to a third-party product. All kinds of issues with not being able to set connect strings or DbProviders due to the dll not referencing the config. There are work-arounds but they're messy and annoying - eg as having to deploy the SQLite dlls to the parant application's folder and manipulating the machine.config programmatically. Blech. (Why Microsoft won't allow class libraries to (optionally) refer to their own config at run time escapes me and I think quite a few other people as there are definitely times when it would be preferable.) These problems are not specific to SQLite/Data Entity/System.Data.SQLite, btw. </p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Vlada Buba** on September 23, 2013 4:34_

<p>Great thanks for the great start up example! Now I'm learning EF from Julie Lerman's "Learning Entity Framework"</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Mayank Gupta** on October 8, 2013 18:26_

<p>This comment has been removed by the author.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Mayank Gupta** on October 8, 2013 23:27_

<p>I have created a new database file. Using that I'am getting <br>System.Data.Entity.ModelConfiguration.ModelValidationException<br><br>Is it because if the wrong use of the types in my model classes. Is there any way to see the .sqlite file. I have an sqlite  editor but it is not opening the database file</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **SarShaGaming** on October 11, 2013 9:26_

<p>Hey Brice.<br><br>Say i want to use another database.<br>The classes i create need to be identical to the database structure?<br><br>EF will connect each property to the table field by the name?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **SarShaGaming** on October 11, 2013 9:27_

<p>This comment has been removed by the author.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on November 6, 2013 15:13_

<p>I use with sqlite a free tool Valentina Studio at the moment. Does everything you need, and does it very well. <a href="http://www.valentina-db.com/en/valentina-studio-overview" rel="nofollow noopener" title="http://www.valentina-db.com/en/valentina-studio-overview">http://www.valentina-db.com...</a></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Mayank Gupta** on November 12, 2013 17:06_

<p>Foriegn Key relation doesn't work. When I do Artist.Albums, it throws ObjectDisposedException</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on February 17, 2014 0:46_

<p>Hi Brice,<br>Thanks for the article.<br><br>But I am getting below exception.<br><br>The type initializer for 'System.Data.Entity.Internal.AppConfig' threw an exception.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on March 12, 2014 2:10_

<p>all the async extensions seem to throw "InvalidCastException".<br>E.g.:<br>await Context.Customers.LoadAsync();<br><br>is there a workaround? thanks.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Anonymous** on April 19, 2014 0:01_

<p>Could you please add a new example of how to accomplish this with the updated versions? I have spent hours trying and keep getting various errors. Or would you atleast be able to post the full working App.Config file? As I believe that is where all the errors are coming from.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[hfgh35jdf353](https://disqus.com/by/bobthe64kmperdaycommuter)** on September 8, 2014 17:36_

<p>Excellent choice on  context.Artists.Remove(avril);</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Clanger Buche](https://disqus.com/by/clangerbuche)** on November 9, 2014 9:19_

<p>I have been predicting that they would add something stupid like that.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Nerissa](https://disqus.com/by/disqus_S30R7xWy1V)** on January 27, 2015 5:30_

<p>could I create this with C sharp???</p><p><a href="http://ddlsoftwarez.blogspot.com/" rel="nofollow noopener" title="http://ddlsoftwarez.blogspot.com/">FREE SOFTWARE DOWNLOAD</a> | <a href="http://ddlsoftwarez.blogspot.com/" rel="nofollow noopener" title="http://ddlsoftwarez.blogspot.com/">SOFTWARE FREE DOWNLOAD</a></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Eldar Zeynalov](https://disqus.com/by/eldar_zeynalov)** on September 18, 2015 5:52_

<p>Good work but i have a problem that it creates sql query with schema names. for ex. <br>SELECT [Extent1].[id] AS [id],  [Extent1].[Content] AS [Content] FROM [dbo].[Posts] AS [Extent1]</p><p>But sqlite does not support (recognize [dbo].[Posts] as Posts table by default) such object names. Can you solve this please ?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **Peter** on September 18, 2015 11:41_

<p>Hi,<br>I have followed the instructions of this article for vs2013 express Update 4 (.Net Framework 4.5) on win7 32bit, but I've got this error when i run the example for first time:</p><p>An unhandled exception of type 'System.InvalidOperationException' occurred in EntityFramework.dll<br>Additional information: No Entity Framework provider found for the ADO.NET provider with invariant name 'System.Data.SQLite'. Make sure the provider is registered in the 'entityFramework' section of the application config file. See <a href="http://go.microsoft.com/fwlink/?LinkId=260882" rel="nofollow noopener" title="http://go.microsoft.com/fwlink/?LinkId=260882">http://go.microsoft.com/fwl...</a> for more information.</p><p>AppConfig:</p><p>&lt;configuration&gt;<br>   &lt;startup&gt; <br>     &lt;supportedruntime version="v4.0" sku=".NETFramework,Version=v4.5"/&gt;<br>   &lt;/startup&gt;<br>   &lt;<a href="http://system.data" rel="nofollow noopener" title="system.data">system.data</a>&gt;<br>      &lt;dbproviderfactories&gt;<br>        &lt;add name="SQLite Data Provider" invariant="System.Data.SQLite" description="Data Provider for SQLite" type="System.Data.SQLite.SQLiteFactory, System.Data.SQLite"/&gt;<br>      &lt;/dbproviderfactories&gt;<br>   &lt;/<a href="http://system.data" rel="nofollow noopener" title="system.data">system.data</a>&gt;<br>   &lt;connectionstrings&gt;<br>      &lt;add name="ChinookContext" connectionstring="Data Source=|DataDirectory|Chinook_Sqlite_AutoIncrementPKs.sqlite" providername="System.Data.SQLite"/&gt;<br>   &lt;/connectionstrings&gt;<br>&lt;/configuration&gt;</p><p>What am i doing wrong ?</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **Peter** on September 18, 2015 13:43_

<p>I finally found a way to make it work with .net 4.5, vs2013, entity framework 6.</p><p>This link <a href="https://github.com/JVimes/LinqSqLiteExample" rel="nofollow noopener" title="https://github.com/JVimes/LinqSqLiteExample">https://github.com/JVimes/L...</a> by JVimes have an example project which works with my configuration and instructions to how to make a new project to work with it.</p><p>my 10x to JVimes :)</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Sandeepkumar Parekh](https://disqus.com/by/sandeepparekh)** on July 14, 2016 7:49_

<p>Thank you</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[HABIB UL REHMAN](https://disqus.com/by/habib_ul_rehman)** on September 1, 2016 10:48_

<p>Hi I used your technique to use entity framework with sqlite but its not working correctly.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by Nishant102-Saurabh** [comment]
Very informative and well-explained tutorial.The sqlite-tutorial concepts are easy to understand, and the examples are helpful for beginners.Great learning resource <a https://www.tpointtech.com/sqlite-tutorial">Learn sqlite-tutorial</a>.
